src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -212,10 +212,18 @@ public void Refresh(IEnumerable<string> affectedFilePaths)
                 foreach (var factory in factories)
                 {
                     var snapshot = factory.CurrentSnapshot;
-                    
-                    if (snapshot.FilesInSnapshot.Any(snapshotPath => affectedFilePaths.Any(affected => PathHelper.IsMatchingPath(snapshotPath, affected))))
+                    var isSnapshotAffected = snapshot.FilesInSnapshot.Any(snapshotPath => affectedFilePaths.Any(affected => PathHelper.IsMatchingPath(snapshotPath, affected)));
+
+                    if (isSnapshotAffected)
                     {
-                        snapshot.IncrementVersion();
+                        if (snapshot.HasNonNavigableIssues)
```

Comment:
Offline discussion:
* create new "GetUpdatedSnapshot" method on snapshot that handles increment/remove non-navigable items as appropriate
* always call factory.UpdateSnapshot(...)
* ?remove IncrementSnapshot?
* HasNonNavigableIssues doesn't need to be public